File: Service/src/main/java/contestUtils/IMainObserver.java
package contestUtils;

import DTO.EventDTO;
import example.model.Participant;

import java.util.List;

public interface IMainObserver {
    void participantAdded(Participant participant) throws Exception;

    void newRegistration(List<EventDTO> events) throws Exception;

    void eventAdded(EventDTO event) throws Exception;
}

File: Service/IMainObserver.cs
using mpp_proiect_csharp_DianaGliga11.Model;
using mpp_proiect_csharp_DianaGliga11.Model.DTO;

namespace Service;

public interface IMainObserver
{
       void ParticipantAdded(Participant participant);
       void EventEvntriesAdded(List<EventDTO> events);
}